<!--
  Copyright (C) 2022, 2023 Shadow Robot Company Ltd <software@shadowrobot.com>

  This program is free software; you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation; either version 3 of the License, or
  (at your option) any later version.

  This program is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program; if not, write to the Free Software
  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
-->

<launch>
  <arg name="rosbag_path" default="$(find polhemus_ros_driver)/config/polhemus_movements.bag"/>
  <arg name="tf_topic_name" default="/tf_polhemus"/>
  <arg name="bimanual" default="false"/>
  <arg name="polhemus_base_right_tf_name" default="polhemus_base_0"/>
  <arg name="polhemus_base_left_tf_name" default="polhemus_base_1"/>
  <arg name="testing_in_isolation" default="false"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="knuckle_to_polhemus_base_right_tf_broadcaster" args="-0.05 0 0.025 1 0 0 0 rh_mf_knuckle_glove $(arg polhemus_base_right_tf_name)" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="knuckle_glove_right_tf_broadcaster" args="0.01 0 0.1 -0.5 -0.5 -0.5 0.5 rh_palm rh_mf_knuckle_glove" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="knuckle_to_polhemus_base_tf_broadcaster" args="-0.05 0 0.025 1 0 0 0 lh_mf_knuckle_glove $(arg polhemus_base_left_tf_name)" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="knuckle_glove_tf_broadcaster" args="0.01 0 0.1 -0.5 -0.5 -0.5 0.5 lh_palm lh_mf_knuckle_glove" />

  <group if="$(arg testing_in_isolation)">
    <include file="$(find polhemus_ros_driver)/launch/polhemus_world_static_transform.launch"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="polhemus_base_to_base_0_tf_broadcaster" args="0 0 0.5 0 0 3.141592653589793 polhemus_base rh_palm" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="polhemus_base_to_base_1_tf_broadcaster" args="0 0 0.5 0 0 3.141592653589793 polhemus_base lh_palm" />
    <!-- by including the substring 'polhemus_tf_broadcaster' in the following static_transform_publisher...
         we can trick the glove calibration gui into thinking the polhemus  driver has started -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="world_to_polhemus_world_polhemus_tf_broadcaster" args="0 0 0.5 0 0 3.141592653589793 world polhemus_world" />
  </group>

  <include file="$(find sr_utilities_common)/launch/real_time_tf_republisher.launch">
    <arg name="tf_topic_name" value="$(arg tf_topic_name)"/>
    <arg name="rosbag_path" value="$(arg rosbag_path)"/>
    <arg name="tf_name_regexes" default="[polhemus]"/>
  </include>

</launch>
